common/djangoapps/student/views.py

```diff
@@ -470,6 +470,10 @@ def signin_user(request):
 def register_user(request, extra_context=None):
     """Deprecated. To be replaced by :class:`student_account.views.login_and_registration_form`."""
     # Determine the URL to redirect to following login:
+    if 'ALLOW_PUBLIC_ACCOUNT_CREATION' in settings.FEATURES and \
+            settings.FEATURES["ALLOW_PUBLIC_ACCOUNT_CREATION"]:
+        return Http404('<h1>Page not found</h1>')
```

Comment:
You can use compact variant `raise Http404`

Reply:
Done